DevelopersGuide.md

```diff
@@ -0,0 +1,192 @@
+  - `--namespace` to change the namespace to install Syndesis into
+  - `--hostname` to specify an hostname for your vm
+
+##### Bring up minishift console
+- `minishift console` or `minishift status`
+
+## Work on a localy copy of syndesis-rest
+- from `syndesis-rest/README.md`:
+```bash
+# create image
+eval $(minishift docker-env)
+cd runtime
+# build code and produce corresponding docker images
+mvn fabric8:build -Dfabric8.mode=kubernetes
+
+# now redploy the rest pod (either from cli or from web)
+oc rollout latest dc/syndesis-rest
+```
+
+##### Collection of helper scripts to automaticlaly build locally and deploy to minishift
+- they build AND deploy
+- in `syndesis-project/tools/minishift-build/`:
+```bash
+atlasmap.sh
+build-all.sh
+integration-runtime.sh
+README.md
+rest.sh
+ui.sh
+vars.sh
+verifier.sh
+```
+
+
+
+## Web development
+- based on the idea of modify a service (route) definition in Syndesis Openshift config, so that it will route the invocation to the local dev webeserver running on host
+- this happens in this script: `scripts/minishift-setup.sh`
+0. install `yarn`
+```bash
+sudo dnf config-manager --add-repo https://dl.yarnpkg.com/rpm/yarn.repo
+sudo dnf install -y yarn
+```
+1. build `syndesis-ui`
+```
+cd syndesis-ui
+yarn install
+cd scripts
+sh minishift-setup.sh
+sh syndesis-install
+```
+2. in another terminal
+```
+cd syndesis-ui
+yarn start:minishift
+# takes a while
+```
+3. (eventually) run tests
+```
+yarn test
+```
+4. Smoke test for development
+```
+# edit src/scss/_overrides.scss and add some flashy rule, like
+* { background: red; }
+# save and refresh the browser window
+
+```
+
+
+## Verifier
+- same worfklow as rest layer if you have to test it in integration with everything else **BUT..**
+- it works well in standalone, doesn't really requires to talk to any other service. It can then be run in full isolation on the host, either from within the IDE or from commandline with `mvn springboot:run` and you can interact with it with plain unauthenticated `curl` invocations.
+
+## Atlasmap
+
+##### In standalone mode:
+```bash
+git clone https://github.com/atlasmap/atlasmap-ui.git
+git clone https://github.com/atlasmap/atlasmap-runtime.git
+cd atlasmap-runtime
+mvn clean install -DskipTests
+cd runtime/
+# itests profile enables debug mode, listining on port 8000 for a java debugger, and changes the default port so that atlasmap-ui standalone can connect to it
+mvn -Pitests spring-boot:run
+
+# leave the runtime running, and in another shell
+cd ../../
+cd atlasmap-ui
+yarn install ; yarn start
+# it will open a browser tab, connected to the standalone runtime you can interact with
+# it will save the corresponding xml mapping file in `atlasmap/atlasmap-runtime/runtime/target/mappings/` where you can inspect the produced file 
+```
+
+###### In full integration with Syndesis
+- ends up using the steps for Web Development 
+```bash
+# after you have built atalsmap-ui project with yarn
```

Comment:
Actually no need to build atlasmap-ui at all, this occurs when you build the main syndesis project.